staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -273,20 +287,46 @@ func (s *SpdyRoundTripper) tlsConn(ctx context.Context, rwc net.Conn, targetHost
 // dialWithoutProxy dials the host specified by url, using TLS if appropriate.
 func (s *SpdyRoundTripper) dialWithoutProxy(ctx context.Context, url *url.URL) (net.Conn, error) {
 	dialAddr := netutil.CanonicalAddr(url)
-	dialer := s.Dialer
-	if dialer == nil {
-		dialer = &net.Dialer{}
+	dial := s.DialContext
+	if dial == nil && s.Dialer != nil {
+		dial = s.Dialer.DialContext
+	} else if dial == nil {
+		nd := &net.Dialer{}
+		dial = nd.DialContext
```

Comment:
maybe this way is more intuitive?

```suggestion
	dial := s.DialContext
	if dial == nil {
	    if s.Dialer != nil {
		dial = s.Dialer.DialContext
        	} else {
		  nd := &net.Dialer{}
		dial = nd.DialContext
		}
	}
```

Reply:
Seems fine, updating.